pkg/cli/cli_options.go

```diff
@@ -0,0 +1,128 @@
+package cli
+
+import (
+	"fmt"
+	"github.com/klothoplatform/klotho/pkg/cli_config"
+	"github.com/klothoplatform/klotho/pkg/yaml_util"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"gopkg.in/yaml.v3"
+	"io/fs"
+	"os"
+)
+
+const configFileName = "options.yaml"
+
+type (
+	Options struct {
+		Update UpdateOptions `yaml:",omitempty"`
+	}
+
+	UpdateOptions struct {
+		Stream string `yaml:",omitempty"`
+	}
+)
+
+func ReadOptions() (Options, error) {
+	var options Options
+	_, fileContents, err := readOptionsFileBytes()
+	if err != nil {
+		return options, err
+	}
+	err = yaml.Unmarshal(fileContents, &options)
+	return options, err
+}
+
+func SetOptions(options map[string]string) error {
+	if len(options) == 0 {
+		// This isn't just an optimization. If the existing file is invalid, then we don't want an error message coming
+		// from this path (since the user hasn't specified options to write, and would be confused by a message saying
+		// "couldn't write CLI options" or similar)
+		return nil
+	}
+	filePath, optionsYaml, err := readOptionsFileBytes()
+	if err != nil {
+		return err
+	}
+	optionsYaml, err = setOptions(optionsYaml, options)
+	if err != nil {
+		return nil
+	}
+
+	err = os.WriteFile(filePath, optionsYaml, 0600)
+	if err != nil {
+		err = errors.Wrap(err, "couldn't write CLI options file")
+	}
+	return err
+}
+
+// setOptions inserts the given options into the yaml, validating along the way that the options still form a valid
+// Options.
+func setOptions(optionsYaml []byte, options map[string]string) ([]byte, error) {
+	logger := zap.S()
+	// First, a warning if the original file isn't valid
+	if err := yaml_util.CheckValid[Options](optionsYaml, yaml_util.Lenient); err != nil {
+		return nil, errors.Wrap(err, "existing options file is invalid")
+	} else if warns := yaml_util.CheckValid[Options](optionsYaml, yaml_util.Strict); warns != nil {
+		logger.Warn(`Existing options contain extra parameters:`)
+		for _, e := range yaml_util.YamlErrors(warns) {
+			logger.Warnf(`▸ %s`, e)
```

Comment:
why only warn instead of raising actual errors? Wouldnt this signal an incorrect usage of the set-options flag?

Reply:
It _probably_ signals incorrect usage, yes. But the incorrectness is just that it won't have any effect, not that it will have the wrong effect. So I figure a nice yellow warning is sufficient for that.

From previous experience, I've found that occasionally you want the ability to set a config that doesn't yet exist, which will take effect on the next update. For example, the next version of the tool may have an automatic migration that you want to modify. In cases like this, being able to specify a not-yet-existent config can be very useful. This approach also means that if you temporarily downgrade your version (once we have version pinning), you won't break your options file if you had set an option in 0.5.21 that doesn't exist in the 0.5.20 you've downgraded to.

Basically, I've found that being strict about "no unknown options" can get people into sticky situations, and that a nice clear warning is sufficient for catching typos.